### PR TITLE
Fix more warnings

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Comment by joshuaboniface** [comment]
Looks like this is failing to build @Bond-009

File: Emby.Naming/Video/VideoListResolver.cs [hunk 1]
```diff
@@ -213,9 +209,9 @@ private bool IsEligibleForMultiVersion(string folderName, string testFilename)
             if (testFilename.StartsWith(folderName, StringComparison.OrdinalIgnoreCase))
             {
                 testFilename = testFilename.Substring(folderName.Length).Trim();
-                return string.IsNullOrEmpty(testFilename) ||
-                       testFilename.StartsWith("-") ||
-                       string.IsNullOrWhiteSpace(Regex.Replace(testFilename, @"\[([^]]*)\]", string.Empty)) ;
+                return string.IsNullOrEmpty(testFilename)
+                    || testFilename[0] == '-'
```
Comment (DrPandemic) [inline Emby.Naming/Video/VideoListResolver.cs]: That's a change in behavior. I'm really unfamiliar with what `IsEligibleForMultiVersion` is. Just wanted to double check that this is the expected behavior.

File: Emby.Naming/TV/EpisodePathParser.cs [hunk 2]
```diff
@@ -109,17 +102,15 @@ private static EpisodePathParserResult Parse(string name, EpisodeExpression expr
...
-                            result.Success = true;
-                        }
+                        result.Year = date.Year;
+                        result.Month = date.Month;
+                        result.Day = date.Day;
+                        result.Success = true;
                     }
 
+
                     // TODO: Only consider success if date successfully parsed?
```
Comment (dkanada) [inline Emby.Naming/TV/EpisodePathParser.cs]: Did you mean to add a second newline here?

File: Emby.Naming/TV/SeasonPathParser.cs [hunk 3]
```diff
@@ -54,27 +48,31 @@ public SeasonPathParserResult Parse(string path, bool supportSpecialAliases, boo
...
         {
-            var filename = Path.GetFileName(path);
+            var filename = Path.GetFileName(path) ?? string.Empty;
 
             if (supportSpecialAliases)
             {
                 if (string.Equals(filename, "specials", StringComparison.OrdinalIgnoreCase))
                 {
-                    return new Tuple<int?, bool>(0, true);
+                    return (0, true);
```
Comment (dkanada) [inline Emby.Naming/TV/SeasonPathParser.cs]: I think the old syntax was more readable but I guess they're both fine.

File: Emby.Naming/Video/Format3DParser.cs [hunk 4]
```diff
@@ -15,10 +15,12 @@ public Format3DParser(NamingOptions options)
 
         public Format3DResult Parse(string path)
         {
-            var delimeters = _options.VideoFlagDelimiters.ToList();
-            delimeters.Add(' ');
+            int oldLen = _options.VideoFlagDelimiters.Length;
+            var delimeters = new char[oldLen + 1];
+            _options.VideoFlagDelimiters.CopyTo(delimeters, 0);
+            delimeters[oldLen] = ' ';
```
Comment (dkanada) [inline Emby.Naming/Video/Format3DParser.cs]: Is the efficiency worth it?

Comment (Bond-009) [reply Emby.Naming/Video/Format3DParser.cs]: I don't have any benchmarks atm, but considering that the old code would allocate at least 2 times (3 times in worst case scenarios) while the new code only allocates once, I'd say it's worth it. (Depending on how hard you want to micro-optimize 😉)

**Pull request merged**